heaps/min_heap.py

```diff
@@ -10,67 +10,105 @@ def __str__(self):
     def __repr__(self):
         return str(self.value)
 
+
 class MinHeap:
 
     def __init__(self):
         self.store = []
 
-
-    def add(self, key, value = None):
+    def add(self, key, value=None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: Ologn
+            Space Complexity: On
         """
-        pass
+        if value is None:
+            value = key
+
+        node = HeapNode(key, value)
+
+        self.store.append(node)
+        index = len(self.store) - 1
+
+        self.heap_up(index)
 
     def remove(self):
         """ This method removes and returns an element from the heap
             maintaining the heap structure
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: Ologn
+            Space Complexity: O1
```

Comment:
✨ Because of the recursive call stack of `heap_down`, space complexity is O(log n)